core/src/main/java/org/apache/druid/java/util/metrics/JvmMonitor.java

```diff
@@ -183,64 +169,110 @@ private GcCounters tryCreateGcCounters()
    */
   private class GcCounters
   {
-    private final List<GcGeneration> generations = new ArrayList<>();
+    private final List<GcCollector> generations = new ArrayList<>();
 
     GcCounters()
```

Comment:
Couple questions about the new mechanism:

1. Do we still need `--add-exports java.base/jdk.internal.perf=ALL-UNNAMED` on Java 11+ after these changes?
2. Can `new GcCounters()` still fail on some JVM versions or configurations?

If either of these is "no" then we should adjust the behavior of `tryCreateGcCounters`.

Reply:
For the first question, I'm 100% sure that this JVM argument is no longer needed. Because the code in my repo supports JDK11 and plus, and I have never added such flag.

Reply:
This patch is tested on jdk11 and jdk15.  Both `--add-exports java.base/jdk.internal.perf=ALL-UNNAMED`  and `tryCreateGcCounters` are no longer needed